tests/4.5/target_teams_distribute/test_target_teams_distribute_default_firstprivate.F90

```diff
@@ -0,0 +1,61 @@
+!===--- test_target_teams_distribute_default_private.F90--------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses the default firstprivate clause on a target teams distribute
+! directive to indicate that a scalar variable should be by default made
+! firstprivate to each team executing the teams distribute region.  The test
+! then operates on the firstprivatized variable in such a way that would most
+! likely cause competiting operations if the variable is not privatized, and
+! such that the outcome depends on the initial host value of the variable. If
+! the computation completes without errors, we assume that the
+! firstprivatization occured.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_target_teams_distribute_device
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+  INTEGER :: errors
+  errors = 0
+  OMPVV_TEST_OFFLOADING
+  OMPVV_TEST_VERBOSE(test_default_firstprivate() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+CONTAINS
+  INTEGER FUNCTION test_default_firstprivate()
+    INTEGER,DIMENSION(N):: a, b, c, d
+    INTEGER:: privatized, errors, x, y
+    errors = 0
+
+    DO x = 1, N
+       a(x) = 1
+       b(x) = x
+       c(x) = 2*x
+       d(x) = 0
+    END DO
+
+    privatized = 5
+
+    !$omp target teams distribute default(firstprivate) shared(a, &
+    !$omp& b, c, d)
+    DO x = 1, N
+       DO y = 1, a(x) + b(x)
+          privatized = privatized + 1
+       END DO
+       d(x) = c(x)*privatized
+    END DO
+
+    DO x = 1, N
+       OMPVV_TEST_AND_SET(errors, (d(x) .ne. (2*x)*(6 + x)))
```

Comment:
check the math.
Looks like the final check does not account for the incremented value of privatized in the last iteration.

Reply:
I believe this math is correct since in the last iteration privatized will be incremented 1 + N times, setting the value to 5 + 1 + N = 6 + N in the last iteration before being multiplied by 2*N on line 52 to give 2*N*(6 + N) as the final value of d(x).

Reply:
Add num_teams()

Reply:
Add num_teams, num_threads and thread_limit if you want to be sure. You need to catch the catch the case where only one team is created.

Reply:
I tried these suggestions and there were the results:
Adding a privatized = 5 after line 52: PASS
Adding a privatized = 5 after line 48: PASS
Adding num_teams(1): FAIL
Adding num_teams(1) and thread_limit(1): FAIL

This suggests to me that this tests exposes a bug, but I will continue to try other changes.

Reply:
Try with more than 1 team

Reply:
With some more investigation I think I've figured it out. Looks like `firstprivate` is functioning at the teams level, not the iteration level. This means that `privatized` is initialized only for the first iteration that each team gets. So the test isn't compliant as written, but by adding a `privatized = 5` at the end of each iteration the test passes all compilers and I believe is compliant and tests what we expect.